.github/workflows/tests.yml

```diff
@@ -325,6 +325,44 @@ jobs:
         if: failure() && github.repository_owner == 'deepset-ai' && github.ref == 'refs/heads/main'
 
 
+  integration-tests-weaviate:
+    name: Integration / Weaviate / ${{ matrix.os }}
+    needs:
+     - unit-tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    services:
+      weaviate:
+        image: semitechnologies/weaviate:1.16.0
```

Comment:
What about the hard-coded version semitechnologies/weaviate:1.14.1 in the fixture in conftest? https://github.com/deepset-ai/haystack/blob/3319ef6d1c8f0b8a4d80d0f531c6ea0fd1c02e6e/test/conftest.py#L496

Reply:
And the utils: https://github.com/deepset-ai/haystack/blob/main/haystack/utils/doc_store.py#L73